src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddressTest.java

```diff
@@ -0,0 +1,99 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import org.junit.jupiter.api.Test;
+
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+
+class PerformerNameAndAddressTest {
+
+    private final PerformerNameAndAddress performingOrganisationNameAndAddress = PerformerNameAndAddress.builder()
+        .performingOrganisationName("LONDON CITY HOSPITAL")
+        .build();
+
+    private final PerformerNameAndAddress performerNameAndAddress = PerformerNameAndAddress.builder()
+            .identifier("A2442389")
+            .code(HealthcareRegistrationIdentificationCode.CONSULTANT)
+            .performerName("DR J SMITH")
+            .build();
+
+    @Test
+    void when_edifactStringDoesNotStartWithCorrectKey_expect_illegalArgumentExceptionIsThrown() {
+        assertThatThrownBy(() -> PerformerNameAndAddress.fromString("wrong value"))
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Can't create PerformerNameAndAddress from wrong value");
+    }
+
+    @Test
+    void when_edifactStringIsPassed_expect_returnAPerformingOrganisationNameAndAddressObject() {
+        assertAll(
+            () -> assertThat(performingOrganisationNameAndAddress)
+                    .usingRecursiveComparison()
+                    .isEqualTo(PerformerNameAndAddress.fromString("NAD+SLA+++LONDON CITY HOSPITAL")),
+
+            () -> assertThat(performerNameAndAddress)
+                    .usingRecursiveComparison()
+                    .isEqualTo(PerformerNameAndAddress.fromString("NAD+SLA+A2442389:902++DR J SMITH"))
```

Comment:
This method name is "expect return a *Performing Organisation*" but this looks like it's not a performing organisation. I suggest we keep separate tests separate.

Reply:
Split this into a separate test in 4bbb811